solutions/usaco-1160.mdx

```diff
@@ -0,0 +1,153 @@
+---
+id: usaco-1160
+source: USACO Silver 2021 December
+title: Convoluted Intervals
+author: Aadit Ambadkar
+---
+
+[Official Analysis](http://usaco.org/current/data/sol_prob3_silver_dec21.html)
+
+## Alternate Solution
+An alternate solution to this problem is to use complementary counting. Consider a given value of $k$. 
+Count the number of pairs of intervals $i$ and $j$ which satisfy $a_i + a_j > k$ or $b_i + b_j < k$. It is clear that 
+any pair that satisfies $a_i + a_j > k$ cannot satisfy $b_i + b_j < k$, and vice versa. So, these two are 
+disjoint, and we can simply count the number of pairs which satisfy $a_i + a_j > k$ and add that to the 
+number of pairs which satisfy $b_i + b_j < k$. We subtract $n$ from this sum, and that gives the answer for
+that specific value of $k$. This solution is presented below.
+
+<LanguageSection>
+<CPPSection>
+```cpp
+// Created by Aadit Ambadkar
+	
+#include<bits/stdc++.h>
+using namespace std;
+typedef long long ll;
+
+int main() {
+	int n, m; cin >> n >> m;
+	vector<pair<int, int>> v;
+	for (int i = 0; i < n; i++) {
+		int a, b; cin >> a >> b;
+		v.push_back({a, b});
+	}
+	for (int k = 0; k <= 2 * m; k++) {
+		ll acount = 0;
+		ll bcount = 0;
+		for (auto p : v) {
+			for (auto q : v) {
+				if (p.first + q.first > k) acount++;
+				else if (p.second + q.second < k) bcount++;
+			}
+		}
+		ll total = (ll) n * (ll) n;
+		cout << total - acount - bcount << "\n";
+	}
+}
+```
+</CPPSection>
+</LanguageSection>
+
+As the aboves stands, the solution runs in $\mathcal{O}(M\cdot N^2)$. This is enough to pass the first two
+test cases. To pass the first five, we make some optimizations. Create two arrays, `asum` and `bsum`.
+These arrays will store the number of pairs of intervals which sum to a given value, that is:
+
+`asum[x] = # of pairs i and j which satisfy a_i + a_j = x`
+
+`bsum[x] = # of pairs i and j which satisfy b_i + b_j = x`
+
+We can iterate over all possible pairs, and update `asum` and `bsum` respectively.
+Then, we can iterate over each possible $k$, and then over `asum` and `bsum` to find the answer for that $k$.
+This solution is presented below:
+
+<LanguageSection>
+<CPPSection>
+```cpp
+// Created by Aadit Ambadkar
+	
+#include<bits/stdc++.h>
+using namespace std;
+typedef long long ll;
+
+int main() {
+    int n, m; cin >> n >> m;
+    vector<pair<int, int>> v;
+    for (int i = 0; i < n; i++) {
+        int a, b; cin >> a >> b;
+        v.push_back({a, b});
+    }
+    vector<ll> asum(2*m+1, 0);
+    vector<ll> bsum(2*m+1, 0);
+	for (auto p : v) {
+		for (auto q : v) {
+			asum[p.first + q.first]++;
+			bsum[p.second + q.second]++;
+		}
+	}
```

Comment:
1. messed up indentation
2. plz no auto

Reply:
indentation seems fine.... check the raw file its a tab

Reply:
also auto is fine why no auto

Reply:
i agree auto is fine

Reply:
<img width="127" alt="image" src="https://user-images.githubusercontent.com/55369003/147584815-99c2331e-0d7b-42e3-a258-95f2ebf309cb.png">
so that was a lie

Reply:
Huh? I know it looks weird but on my end the indentation is fine...

Reply:
It's not, there's a mix of spaces and tabs.

Reply:
^

Reply:
ok i fixed the issue now